pygenometracks/tracks/BedTrack.py

```diff
@@ -419,40 +421,73 @@ def plot(self, ax, chrom_region, start_region, end_region):
                 self.counter += 1
                 bed = region.data
 
+                if ax.get_xlim()[0] < ax.get_xlim()[1]:
+                    bed_left = bed.start
+                    bed_right = bed.end
+
+                    def add_to_right(a, b):
+                        return(a + b)
+
+                    def add_to_left(a, b):
+                        return(a - b)
+
+                    def is_left_to(a, b):
+                        return(a < b)
+
+                    def is_right_to(a, b):
+                        return(a > b)
+
+                else:
+                    bed_left = bed.end
+                    bed_right = bed.start
+
+                    def add_to_right(a, b):
+                        return(a - b)
+
+                    def add_to_left(a, b):
+                        return(a + b)
+
+                    def is_left_to(a, b):
+                        return(a > b)
+
+                    def is_right_to(a, b):
+                        return(a < b)
+
                 if self.properties['labels']:
                     num_name_characters = len(bed.name) + 2
                     # +2 to account for a space before and after the name
-                    bed_extended_end = int(bed.end + (num_name_characters * self.len_w))
+                    bed_extended_right = int(add_to_right(bed_right, (num_name_characters * self.len_w)))
                 else:
-                    bed_extended_end = (bed.end + 2 * self.small_relative)
+                    bed_extended_right = add_to_right(bed_right, 2 * self.small_relative)
 
+                bed_extended_left = bed_left
                 # get smallest free row
-                start_pos = bed.start
                 if len(row_last_position) == 0:
                     free_row = 0
-                    row_last_position.append(bed_extended_end)
+                    row_last_position.append(bed_extended_right)
                 else:
                     # If all_labels_inside = True
                     # genes which goes over will have their labels inside
                     if self.properties['all_labels_inside'] and self.properties['labels'] \
-                       and bed_extended_end > end_region:
-                        start_pos = int(bed.start - (num_name_characters * self.len_w))
+                       and is_right_to(bed_extended_right, ax.get_xlim()[1]):
+                        bed_extended_left = int(add_to_left(bed_left, (num_name_characters * self.len_w)))
```

Comment:
add_to_left is now always an int? So the cast can be removed?

Reply:
No add_to_left is just doing addition or substraction but I need the cast.